docs: add peering metric doc (#13862)
hashicorp/consul: website/content/docs/agent/telemetry.mdx

| `consul.proxy.web.upstream.conns`   | Shows the current number of connections open from a proxy instance to an upstream. Where supported a `src` label is added indicating the service name the proxy represents, and a `dst` label is added indicating the service name the upstream is connecting to. | connections | gauge   |
| `consul.proxy.web.inbound.rx_bytes` | Increments by the number of bytes received from an upstream connection. Where supported a `src` label is added indicating the service name the proxy represents, and a `dst` label is added indicating the service name the upstream is connecting to.       | bytes       | counter |
| `consul.proxy.web.inbound.tx_bytes` | Increments by the number of bytes transferred to an upstream connection. Where supported a `src` label is added indicating the service name the proxy represents, and a `dst` label is added indicating the service name the upstream is connecting to.      | bytes       | counter |

## Peering metrics

** Requirements: **
* Consul 1.13.0+

> Consul 1.13 introduced [cluster peering](/docs/connect/cluster-peering). These metrics are only emitted by the leader server.

Reviewer: ```suggestion
[Cluster peering](/docs/connect/cluster-peering) refers enabling communication between Consul clusters through a peer connection, as opposed to a federated connection. Consul collects metrics that describe the number of services exported to a peered cluster. Peering metrics are only emitted by the leader server.  
```
1. The Consul version is already stated in the requirements
2. Added a statement about what cluster peering is so readers would have a little more context without having to follow the link.
3. It appears that there is only one metric for now that describes the number of exported services, correct?

> Consul 1.13 introduced [cluster peering](/docs/connect/cluster-peering). These metrics are only emitted by cluster leaders as of now.

Reviewer 2: I would avoid setting expectations about this changing

```suggestion
Consul 1.13 introduced [cluster peering](/docs/connect/cluster-peering). These metrics are only emitted by the leader server.
```

> Consul 1.13 introduced [cluster peering](/docs/agent/connect/cluster-peering#index). These metrics are only emitted by cluster leaders as of now.

Author: This needs to be `/docs/connect/cluster-peering`

| Metric                                | Description                                                           | Unit   | Type    |
| ------------------------------------- | ----------------------------------------------------------------------| ------ | ------- |
| `consul.peering.exported_services`    | Counts the number of services exported to a peer cluster.  | count  | gauge   |

### Labels

> 

Reviewer: ```suggestion
Consul attaches the following labels to metric values. 
```
I thought an introduction might be helpful. Is that how labels for metrics work? Or are labels attributes that the user can assign? Is the concept of labels for metrics described anywhere?

Author: no labels are not user controlled

Author: and I am not entirely sure that our docs explain label based metrics anywhere 🤔  (at least I couldn't find a link right now)

> | `consul.peering.exported_services`    | Counts the number of exported services per peering in the partition.  | count  | gauge   |

Reviewer 2: This might be more direct:
```suggestion
| `consul.peering.exported_services`    | Counts the number of services exported to a peer cluster.  | count  | gauge   |
```

| Label Name                            | Description                                                            | Possible values                            |
| ------------------------------------- | ---------------------------------------------------------------------- | ------------------------------------------ |
| `peer_name`                           | The name of the peering on the reporting cluster/ leader.              | Any defined peer name in the cluster.      |

> | `peer_id`                             | The id of the peering on the reporting cluster/ leader.                | Any UUID.                                  |

Reviewer: ```suggestion
| `peer_id`                             | The ID of the peering on the reporting cluster/ leader.                | Any UUID.                                  |
```
Not that anyone would be confused that we're talking about the psychoanalytical id

> | `partition`                           | Enterprise only: Name of the partition that the peering is created in. | Any defined partition name in the cluster. |

Reviewer 2: We use a special tag for highlighting enterprise features:
```suggestion
| `partition`                           | <EnterpriseAlert inline /> Name of the partition that the peering is created in. | Any defined partition name in the cluster. |
```
